docs/website/docs/reference/explainers/how-dlt-works.md

```diff
@@ -6,33 +6,47 @@ keywords: [architecture, extract, normalize, load]
 
 # How `dlt` works
 
-`dlt` automatically turns JSON returned by any [source](../../general-usage/glossary.md#source)
-(e.g., an API) into a live dataset stored in the
-[destination](../../general-usage/glossary.md#destination) of your choice (e.g., Google BigQuery). It
-does this by first [extracting](how-dlt-works.md#extract) the JSON data, then
-[normalizing](how-dlt-works.md#normalize) it to a schema, and finally [loading](how-dlt-works#load)
-it to the location where you will store it.
+In a nutshell, `dlt` automatically turns data from a number of available [sources](../../general-usage/source) (e.g., an API, a PostgreSQL database, or Python data structures) into a live dataset stored in a [destination](../../general-usage/destination) of your choice (e.g., Google BigQuery, a Deltalake on Azure, or by pushing the data back via reverse ETL). You can easily implement your own sources, as long as you yield data in a way that is compatible with `dlt`, such as JSON objects, Python lists and dictionaries, pandas dataframes, and arrow tables. `dlt` will be able to automatically compute the schema and move the data to your destination.
 
-![architecture-diagram](/img/architecture-diagram.png)
+![architecture-diagram](/img/dlt-onepager.svg)
 
-## Extract
+## A concrete example
 
-The Python script requests data from an API or a similar
-[source](../../general-usage/glossary.md#source). Once this data is received, the script parses the
-JSON and provides it to `dlt` as input, which then normalizes that data.
+The main building block of `dlt` is the [pipeline](../../general-usage/glossary.md#pipeline), which orchestrates the loading of data from your source into your destination in three discrete steps when you call its `run` method. Consider this intentionally short example:
 
-## Normalize
+```py
+import dlt
 
-The configurable normalization engine in `dlt` recursively unpacks this nested structure into
-relational tables (i.e., inferring data types, linking tables to create nested relationships,
-etc.), making it ready to be loaded. This creates a
-[schema](../../general-usage/glossary.md#schema), which will automatically evolve to accommodate any future
-source data changes (e.g., new fields or tables).
+pipeline = dlt.pipeline(pipeline_name="my_pipeline", destination="duckdb")
+pipeline.run([{"id": 1}, {"id": 2}, {"id": 3}], table_name="items")
```

Comment:
maybe add a hint:
Use `progress="log"` to see logs for each stage separately https://dlthub.com/docs/walkthroughs/run-a-pipeline#2-see-the-progress-during-loading

Reply:
I think this is more something for the walkthrough